presto-hudi/src/main/java/com/facebook/presto/hudi/HudiSplitManager.java

```diff
@@ -151,35 +155,7 @@ private ExtendedFileSystem getFileSystem(ConnectorSession session, HudiTableHand
         }
     }
 
-    private Optional<HudiSplit> createHudiSplit(
-            HudiTableHandle table,
-            FileSlice slice,
-            String timestamp,
-            HudiPartition partition,
-            HudiSplitWeightProvider splitWeightProvider)
-    {
-        HudiFile hudiFile = slice.getBaseFile().map(f -> new HudiFile(f.getPath(), 0, f.getFileLen())).orElse(null);
-        if (null == hudiFile && table.getTableType() == HudiTableType.COW) {
-            return Optional.empty();
-        }
-        List<HudiFile> logFiles = slice.getLogFiles()
-                .map(logFile -> new HudiFile(logFile.getPath().toString(), 0, logFile.getFileSize()))
-                .collect(toImmutableList());
-        long sizeInBytes = hudiFile != null ? hudiFile.getLength()
-                : (logFiles.size() > 0 ? logFiles.stream().map(HudiFile::getLength).reduce(0L, Long::sum) : 0L);
-
-        return Optional.of(new HudiSplit(
-                table,
-                timestamp,
-                partition,
-                Optional.ofNullable(hudiFile),
-                logFiles,
-                ImmutableList.of(),
-                NodeSelectionStrategy.NO_PREFERENCE,
-                splitWeightProvider.calculateSplitWeight(sizeInBytes)));
-    }
-
-    private static HudiPartition getHudiPartition(ExtendedHiveMetastore metastore, MetastoreContext context, HudiTableLayoutHandle tableLayout, String partitionName)
+    public static HudiPartition getHudiPartition(ExtendedHiveMetastore metastore, MetastoreContext context, HudiTableLayoutHandle tableLayout, String partitionName)
```

Comment:
is there some way to cache this information? is the partition metadata mostly immutable? Looks like a cache here, that expires every minute or so could really help reduce calls to metastore?

Reply:
that's a good idea and presto provides `CachingHiveMetastore`, which is metastore cache implementation of `ExtendedHiveMetastore`.  Wil add that.

Reply:
Left a comment above. to understand TTL, how invalidation works, and side effects